src/store/keycodes.js

```diff
@@ -77,9 +77,16 @@ function generateKeycodes(osKeyboardLayout, isSteno = false) {
     ...(isSteno ? steno : [])
   ];
   const { keycodeLUT } = keymapExtras[getOSKeyboardLayout()];
-  return keycodes.map((keycodeObject) =>
-    toLocaleKeycode(keycodeLUT, keycodeObject)
-  );
+  return [
+    ...keycodes.map((keycodeObject) =>
+      toLocaleKeycode(keycodeLUT, keycodeObject)
+    ),
+    ...Object.entries(keycodeLUT).map(([code, { name, title }]) => ({
+      code: title?.split(' ')[0],
+      name,
+      title: code
+    }))
+  ];
```

Comment:
We could in theory just add all languages like this, so that one can upload `NO_`, `FR_` and everything else regardless of what language is configured in _Settings_. Would be easier to use, and I don't see any harm except maybe longer searches for match.

Reply:
I'm not sure what the value of that is, given that the interpretation is based on the OS setting. Don't you think this would be more confusing than helpful?

Reply:
Can agree that it's a bit confusing when having multiple languages at the same time, but for that to happen, the user would have to configure it on their own before importing. It could also be useful for some to have multiple languages in different layers if they jump between computers with different host languages.

The nice thing here is that users do not need to remember nor know that they have to to configure host language before importing their NO_ or FR_ keyboard. As they use language-specific codes, they would probably never select the correct host language - because that just gives them the corresponding/mapped KC_ codes, not NO_ or _FR? Thus I think it's nice that the tool actually displays the NO_ and _FR codes correctly with the default host language setting.

I think that is more valuable than potentially confusing those with multiple language codes in the same layout.

Reply:
We could put it behind a feature toggle in the settings. _support extra language keycodes_ or something.

Reply:
What if we display the language codes? Would that help? https://github.com/qmk/qmk_configurator/pull/1374/commits/db616fc0e558481dcc353535f6de857771e1271f

![image](https://github.com/user-attachments/assets/b739511b-25d0-4d9c-9b0d-1fcdd45f799e)

Reply:
I think this will reinforce even more the incorrect preconception that keycodes like `FR_CCED` will invariably produce the "ç" keysym no matter the OS settings because that is what it says on the tin can after all.

It would be nice to properly support JSON files with mixed aliases but the displayed layout should obey the host OS keyboard layout setting or else the setting will lose meaning. So, I'd vote against the display of mixed aliases.

For a keymap like:
```json
["FR_CCED", "IL_MUL", "LSFT_T(NO_T)", "NO_G"]
```
I'd expect these legends under "English (USA)":
```
( 9, RAlt[* 8], LSft[T], G
```
Although one problem with this approach is that there may exist multiple keymap_extras variants for the same language (e.g. `keymap_french.h` and `keymap_french_afnor.h`) and then it becomes ambiguous whether `FR_CCED` should translate to `KC_9` or to `ALGR(KC_C)`.

The `host_language` field in the JSON could potentially help to solve the ambiguity but as far as I can tell that only accepts a single host language and if we're dealing with JSON files mixing aliases from multiple keymap_extra aliases, who knows which host_language will be picked (it might be "norwegian" instead of either "french" or "french_afnor" in which case the translation becomes ambiguous)

Reply:
This is how it's looking in the Zsa configurator btw. First have to enable the language, and can then use it. By having to enable it before the relevant keys are possible to select, the information is at least somewhat conveyed to the user.

![Screenshot_20241113_192022_Chrome.jpg](https://github.com/user-attachments/assets/e4e45901-1c5a-47dd-8bd1-31b7d1d3dab8)

![Screenshot_20241113_191840_Chrome.jpg](https://github.com/user-attachments/assets/18a3f31f-c81f-4663-bc5a-d10993345493)

Reply:
I'm still in the middle of moving to pinia so I'm not sure I want this disruption right now for what is a pretty confusing and niche use case.

Reply:
Not completely comparable as the qmk configurator does not support setting the language-specific keycodes in the GUI. Here, it only applies to keymaps that already contain language-specific key codes on import.

I think the main thing I would like to accomplish here is 1. to not change the keycodes when importing and exporting a keymap if no modifications are made, and 2. ideally somewhat indicate what's actually in the JSON file for that key.

Other than that, I'm very open to making modifications.

Reply:
I can already imagine the case of the German programmer who hates having to use AltGr to access [] and {} and prefers the `[ {` and `] }` keysyms found on US layouts but still wants access to the ß, ü, ä and ö keys she needs to type German. Mixing the US `[ {` keys with the German `ß` key sounds like the perfect solution except it won't work as she expects.

In fact, I remember multiple people on Discord tripping up over pretty much the same thing because they used the default `KC_` keycodes everywhere in their keymap and then imported a keymap_extra just "to get access" to a special key and were confused why the special key they added to their keymap was not producing the expected result.